core/filestorage/filesync/filesync.go

```diff
@@ -87,76 +90,94 @@ func New() FileSync {
 	return &fileSync{}
 }
 
-func (f *fileSync) Init(a *app.App) (err error) {
-	f.dbProvider = app.MustComponent[datastore.Datastore](a)
-	f.rpcStore = a.MustComponent(rpcstore.CName).(rpcstore.Service).NewStore()
-	f.dagService = a.MustComponent(fileservice.CName).(fileservice.FileService).DAGService()
-	f.fileStore = app.MustComponent[filestore.FileStore](a)
-	f.eventSender = app.MustComponent[event.Sender](a)
-	f.removePingCh = make(chan struct{})
-	f.uploadPingCh = make(chan struct{})
+func (s *fileSync) Init(a *app.App) (err error) {
+	s.dbProvider = app.MustComponent[datastore.Datastore](a)
+	s.rpcStore = a.MustComponent(rpcstore.CName).(rpcstore.Service).NewStore()
+	s.dagService = a.MustComponent(fileservice.CName).(fileservice.FileService).DAGService()
+	s.fileStore = app.MustComponent[filestore.FileStore](a)
+	s.eventSender = app.MustComponent[event.Sender](a)
+	s.removePingCh = make(chan struct{})
+	s.uploadPingCh = make(chan struct{})
+	db, err := s.dbProvider.LocalStorage()
+	if err != nil {
+		return
```

Comment:
In case of err here fileSync.Close will panic on queues closing. Maybe check for nil in Close

Reply:
thank you, fixed